### DISCUSSION TITLE
Time format & localtime

### DISCUSSION THREAD

**Discussion by JStyle21** [thinkst/canarytokens#251] [Feature Requests]
Hi,

Is it possible to change the time format and use something other than UTC?
I've found a TIME_FORMAT variable in the email alerts but it's not being used, i've also tried to change the time in 4-5 different files and location but non of them seem to impact the result and there are so many other occurrences, where is the correct place to change it so it's reflected in the emails and dashboard?

**Comment by JStyle21** [comment]
I found 1 place which is the history page it's under httpd_site.py -> HistoryPage -> render_GET -> formatted_timestamp
however no matter how i convert UTC to local all i get is Processing Failed, no idea what's causing this as the new timestamp is valid, same format and everything.

I've tested this locally in a different program without any issue...

**Comment by jayjb** [comment]
Hi @JStyle21,

It's a good idea (to be able to trivially change the time format). I've labelled it as a feature request for now since as you have noted there isn't a clean place to do it currently. 

Do you mind sharing the code you tried? Since the python3 port is almost ready, we won't be adding more code right now. But we can line this request up

**Comment by JStyle21** [comment]
Hi @jayjb 

At first i've tried to convert **UTC** to **Local** but it didn't work but then i actually solved this issue by converting all the UTC functions such as `datetime.utcnow()` to `now()`
It was a hassle, starting from the history page, then i noticed the emails had a separate timestamp etc. so i went and changed all of them...
So that + adding the timezone to the docker files has fixed the issue for me. but yes a better solution would be to easily set it in some global variable instead.
I can do a pull request but i doubt you'd want a solution like that.

As for the conversion function, i took it from [stackoverflow](https://stackoverflow.com/a/19238551), here is the code:

```python
from datetime import datetime
import time

def datetime_from_utc_to_local(utc_datetime):
    now_timestamp = time.time()
    offset = datetime.fromtimestamp(now_timestamp) - datetime.utcfromtimestamp(now_timestamp)
    return utc_datetime + offset
```
